operator/apis/loki/v1/lokistack_types.go

```diff
@@ -323,6 +323,28 @@ type LokiTemplateSpec struct {
 	Ruler *LokiComponentSpec `json:"ruler,omitempty"`
 }
 
+// ClusterProxy is the Proxy configuration when the cluster is behind a Proxy.
+type ClusterProxy struct {
```

Comment:
I believe this is the right struct to support custom proxy configuration an any kubernetes/proxy environment. For k8s/openshift clusters where the cluster provisioning provides the proxy variables per enviroment, we should add an extra field to pick them up, e.g.:

```go
ReadVarsFromEnv bool `json:"readVarsFromEnv,omitempty"`
```

Ref: https://sdk.operatorframework.io/docs/building-operators/golang/references/proxy-vars/#m-docsbuilding-operatorsgolangreferencesproxy-vars